src/client/dfs/duns.c

```diff
@@ -1086,6 +1131,194 @@ duns_create_path(daos_handle_t poh, const char *path, struct duns_attr_t *attrp)
 	return rc;
 }
 
+int
+duns_link_cont(daos_handle_t poh, const char *cont, const char *path)
+{
+	daos_handle_t		coh;
+	daos_prop_t		*prop;
+	struct daos_prop_entry	*entry;
+	daos_pool_info_t	pinfo = {0};
+	daos_cont_info_t	cinfo = {0};
+	daos_cont_layout_t	type;
+	char			pool_str[37], cont_str[37];
+	int			len;
+	char			str[DUNS_MAX_XATTR_LEN];
+	char			type_str[10];
+	int			rc, rc2;
+
+	if (path == NULL) {
+		D_ERROR("Invalid path\n");
+		return EINVAL;
+	}
+
+	rc = daos_pool_query(poh, NULL, &pinfo, NULL, NULL);
+	if (rc) {
+		D_ERROR("Failed to query pool info" DF_RC "\n", DP_RC(rc));
+		return daos_der2errno(rc);
+	}
+
+	rc = daos_cont_open(poh, cont, DAOS_COO_RO, &coh, &cinfo, NULL);
+	if (rc) {
+		D_ERROR("daos_cont_open() failed "DF_RC"\n", DP_RC(rc));
+		return daos_der2errno(rc);
+	}
+
+	prop = daos_prop_alloc(1);
+	if (prop == NULL)
+		D_GOTO(out_cont, rc = ENOMEM);
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+	rc = daos_cont_query(coh, NULL, prop, NULL);
+	if (rc) {
+		daos_prop_free(prop);
+		D_ERROR("daos_cont_query() failed, "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out_cont, rc = daos_der2errno(rc));
+	}
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_LAYOUT_TYPE);
+	if (entry == NULL) {
+		daos_prop_free(prop);
+		D_ERROR("Invalid container type\n");
+		D_GOTO(out_cont, rc = EINVAL);
+	}
+	type = entry->dpe_val;
+	daos_prop_free(prop);
+
+	uuid_unparse(pinfo.pi_uuid, pool_str);
+	uuid_unparse(cinfo.ci_uuid, cont_str);
+
+	if (type == DAOS_PROP_CO_LAYOUT_POSIX) {
+		struct statfs	fs;
+		char		*dir, *dirp;
+		mode_t		mode = S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH;
+		size_t		path_len;
+
+#ifdef LUSTRE_INCLUDE
+		if (fs.f_type == LL_SUPER_MAGIC) {
```

Comment:
Arghhhhh, looks like recent commit ebc3676f2b9876896b66be15473a640805473d35 (DAOS-6540 dfuse: Enable multi-user support in dfuse (#3956)) from @ashleypittman had already removed the statfs() syscall before testing if "fs.f_type == LL_SUPER_MAGIC" in duns_create_path() so you have replicated the error by using cut/paste of this piece of code !!...

Reply:
if not i'll just fix it with this PR

Reply:
just updated this PR to fix

Reply:
> ah ok.. so is there a PR on master that fixes that?

No I just discovered this when doing review for this PR !! Thanks to have fixed it also in duns_create_path().